[Document Intelligence] Release 1.0.0 GA (#31746)
Azure/azure-sdk-for-js: sdk/documentintelligence/ai-document-intelligence-rest/README.md

| SDK version  | Supported API version of service |
| ------------ | -------------------------------- |
| 1.0.0        | 2024-11-30                       |

> | 1.0.0-beta.3 | 2024-07-31-preview               |
> | 1.0.0-beta.2 | 2024-02-29-preview               |
> | 1.0.0-beta.1 | 2023-10-31-preview               |

Reviewer: These should be removed, right?
```suggestion
```

Author: The other languages haven't removed either, and they could be a quick reference for customers too.
I'd keep them.

Reviewer: Preview endpoints are typically deprecated and become offline much sooner than GA ones. When this happens, it would be awkward for our users who download this stable version to still see entries about the gone preview endpoints in the README file.

Author: ok, I'll remove

| Service API version | Supported clients                                            | Package                                                       |
| ------------------- | ------------------------------------------------------------ | ------------------------------------------------------------- |
| 2024-11-30          | DocumentIntelligenceClient                                   | `@azure-rest/ai-document-intelligence` version `1.0.0`        |

> | 2024-07-31-preview  | DocumentIntelligenceClient                                   | `@azure-rest/ai-document-intelligence` version `1.0.0-beta.3` |
> | 2024-02-29-preview  | DocumentIntelligenceClient                                   | `@azure-rest/ai-document-intelligence` version `1.0.0-beta.2` |
> | 2023-10-31-preview  | DocumentIntelligenceClient                                   | `@azure-rest/ai-document-intelligence` version `1.0.0-beta.1` |

Reviewer: ```suggestion
```

// }
```

## Batch analysis

```ts
import {
  parseResultIdFromResponse,

>   AnalyzeResultOperationOutput,

Reviewer: Is this being used?
```suggestion
```

Author: removed

  isUnexpected,
} from "@azure-rest/ai-document-intelligence";

// 1. Analyze a batch of documents
const initialResponse = await client
  .path("/documentModels/{modelId}:analyzeBatch", "prebuilt-layout")
  .post({
    contentType: "application/json",
    body: {
      azureBlobSource: {
        containerUrl: batchTrainingFilesContainerUrl(),
      },
      resultContainerUrl: batchTrainingFilesResultContainerUrl(),
      resultPrefix: "result",
    },
  });

> console.log("initialResponse: ", initialResponse);

Reviewer: Is this needed if we already throwing in the error case and printing in the happy case?

Author: will remove

if (isUnexpected(initialResponse)) {
  throw initialResponse.body.error;
}
const resultId = parseResultIdFromResponse(initialResponse);
console.log("resultId: ", resultId);

> console.log("model id: ", model.modelId);

Reviewer: Is `model` bound? I can't see it.

Author: removed

> // (Optional) You can keep polling for the result which might take longer for the batch analysis which would incur costs.
> //            You can avoid this and can directly skip to retriving the result.
> // const poller = await getLongRunningPoller(client, initialResponse);
> // await poller.pollUntilDone();

Reviewer: fix a typo and rewrite to be clearer:
```suggestion
// (Optional) You can poll for the batch analysis result but be aware that a job may take unexpectedly long time, and polling could incur additional costs.
// const poller = await getLongRunningPoller(client, initialResponse);
// await poller.pollUntilDone();
```

//  }
```

> ## Get analyze PDF

Reviewer: I am not sure if I understand correctly but the title doesn't make sense. Should it be "Get analyed PDF" or just "Analyze PDF"?
```suggestion
## Analyze PDF
```

Author: This is to get the output PDF after an analysis.

Author: I'll call it `Get the generated PDF output from document analysis.`

```ts
const filePath = path.join(ASSET_PATH, "layout-pageobject.pdf");

> const base64Source = fs.readFileSync(filePath, { encoding: "base64" });

Reviewer: nitpick: I think we should show usage of modern APIs such as `fs/promises`.
```suggestion
const base64Source = await fs.readFile(filePath, { encoding: "base64" });
```

const initialResponse = await client
  .path("/documentModels/{modelId}:analyze", "prebuilt-read")
  .post({
    contentType: "application/json",
    body: {
      base64Source,
    },
    queryParameters: { output: ["pdf"] },
  });

if (isUnexpected(initialResponse)) {
  throw initialResponse.body.error;
}

> const poller = await getLongRunningPoller(client, initialResponse, { ...testPollingOptions });

Reviewer: ```suggestion
const poller = await getLongRunningPoller(client, initialResponse);
```

await poller.pollUntilDone();

const output = await client
  .path(
    "/documentModels/{modelId}/analyzeResults/{resultId}/pdf",
    "prebuilt-read",
    parseResultIdFromResponse(initialResponse)
  )
  .get();

> // A PDF's header is expected to be: %PDF-
> assert.ok(output.body.toString().startsWith("%PDF-"));

Reviewer: We shouldn't assert in the README snippets. Could we show an example of streaming the content to disk?
```suggestion
```

```

> ## Get analyze Figures

Reviewer: Same remarks above apply throughout this section too.
